backend/src/main/java/rocks/inspectit/gepard/agentmanager/configuration/service/GitService.java

```diff
@@ -0,0 +1,74 @@
+/* (C) 2024 */
+package rocks.inspectit.gepard.agentmanager.configuration.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.gepard.agentmanager.exception.GitOperationException;
+
+@Slf4j
+@Service
+public class GitService {
+
+  private static final String FILE = "configuration.json";
+
+  @Value("${inspectit-config-server.configurations.local-path}")
+  private String localRepoPath;
+
+  private Git git;
+
+  public void commit() {
+    try {
+      git.add().addFilepattern(FILE).call();
+      git.commit().setMessage("update configuration").call();
+    } catch (GitAPIException e) {
+      throw new GitOperationException("Failed to commit changes to local repository", e);
+    }
+  }
+
+  public void updateFileContent(String content) {
+    try {
+      File file = new File(localRepoPath + "/configuration.json");
+      java.nio.file.Files.write(file.toPath(), content.getBytes());
+    } catch (IOException e) {
+      throw new GitOperationException("Failed to update file content", e);
+    }
+  }
+
+  public String getFileContent() {
+    try {
+      Path filePath = Path.of(localRepoPath, FILE);
+      if (Files.exists(filePath)) {
+        return Files.readString(filePath);
```

Comment:
This is probably not thread-safe. If multiple agents try to read the configuration at the same time, there will be an exception thrown.

Could you add some protection mechanism, so the file can be accessed by multiple threads safely?
For instance, in the agent we use a [FileAccessor](https://github.com/inspectIT/inspectit-gepard-agent/blob/master/inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/internal/file/FileAccessor.java)